package.json

```diff
@@ -1,37 +1,87 @@
 {
   "name": "testcafe-reporter-isaac",
-  "version": "0.1.0",
+  "version": "0.0.0-develop",
+  "author": "ISAAC E-commerce Solutions BV",
+  "license": "MIT",
   "description": "An Allure reporter plugin for TestCafé. Used to generate test reports for e2e-tests.",
-  "main": "index.js",
+  "keywords": [
+    "testcafé",
+    "allure",
+    "integration"
+  ],
+  "main": "index.ts",
   "repository": {
     "type": "git",
     "url": "git+https://github.com/isaaceindhoven/testcafe-reporter-isaac.git"
   },
-  "author": "ISAAC E-commerce Solutions BV",
-  "license": "MIT",
   "bugs": {
     "url": "https://github.com/isaaceindhoven/testcafe-reporter-isaac/issues"
   },
   "homepage": "https://github.com/isaaceindhoven/testcafe-reporter-isaac#readme",
   "scripts": {
+    "build": "tsc",
+    "test": "npm run test:lint && npm run test:unit",
+    "test:lint": "eslint {src,tests}/**/*.{ts,js}",
+    "test:lint:fix": "eslint {src,tests}/**/*.{ts,js} --fix",
+    "test:unit": "jest",
     "version": "node -v",
     "semantic-release": "semantic-release"
   },
   "husky": {
     "hooks": {
+      "pre-commit": "tsc --noEmit && npm run test:unit && lint-staged",
       "prepare-commit-msg": "exec < /dev/tty && git cz --hook || true"
     }
   },
+  "lint-staged": {
+    "*.{js,ts,tsx}": [
+      "eslint --fix"
+    ]
+  },
   "config": {
     "commitizen": {
       "path": "./node_modules/cz-conventional-changelog"
     }
   },
+  "jest": {
```

Comment:
May I suggest adding a `roots`property to the jest configuration with value `['src', 'tests']`?
https://jestjs.io/docs/en/configuration#roots-arraystring

This prevents tests in other folders of being run. For example currently, because of the inclusion of tests in the TS build process, the distributed tests are also picked up by Jest.

Eg.

```
╰─ npm run test:unit

> testcafe-reporter-isaac@0.0.0-develop test:unit /Users/maartenvanhoof/Sites/testcafe-reporter-isaac
> jest

ts-jest[jest-transformer] (WARN) Got a `.js` file to compile while `allowJs` option is not set to `true` (file: /Users/maartenvanhoof/Sites/testcafe-reporter-isaac/dist/tests/index.spec.js). To fix this:
  - if you want TypeScript to process JS files, set `allowJs` to `true` in your TypeScript config (usually tsconfig.json)
  - if you do not want TypeScript to process your `.js` files, in your Jest config change the `transform` key which value is `ts-jest` so that it does not match `.js` files anymore
 PASS  dist/tests/index.spec.js
 PASS  tests/index.spec.ts
-----------|---------|----------|---------|---------|-------------------
File       | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s
-----------|---------|----------|---------|---------|-------------------
All files  |     100 |      100 |     100 |     100 |
 **dist/src**  |     100 |      100 |     100 |     100 |
  index.js |     100 |      100 |     100 |     100 |
 src       |     100 |      100 |     100 |     100 |
  index.ts |     100 |      100 |     100 |     100 |
-----------|---------|----------|---------|---------|-------------------

Test Suites: 2 passed, 2 total
Tests:       2 passed, 2 total
Snapshots:   0 total
Time:        4.317s
Ran all test suites.
```

Reply:
With the spit of the tsconfig files, this should no longer be a problem, however, in the case, the `tsconfig.test.json` is being built via the `tsc` command and the unit tests are run the following roots property has been added to the Jest config:

```
    "roots": [
      "src",
      "tests"
    ],
```

Reply:
Adding it here because it is not possible to directly comment on the config splitting message and this message is related.

Opted to move the configs of the dependencies outside of the `package.json` as suggested. Only with Commitizen this does not seem to be possible and issue exits regarding it but it has not been resolved yet. [ISSUE 460](https://github.com/commitizen/cz-cli/issues/460)

Also most further packages support the .`[packageName]rc.js` format for the config so for consitency this format has been chosen as the default. This does however mean that Jest, the only package not supporting that format (it only supports `jest.config.js` directly), has to make use of a custom config file resulting in the need to change the test command to: `"test:unit": "jest --config .jestrc.js",`.

Reply:
Very, very consistent in the naming of the separate config files, but I personally not go **that** far. If possible, use the package default naming conventions as it might end up making things more complex later on.